runtime/include/gnuradio/concurrent_queue.hpp

```diff
@@ -24,6 +24,21 @@ class concurrent_queue
 
         return true;
     }
+
+    // Non-blocking
+    bool try_pop(T& msg)
+    {
+        std::unique_lock<std::mutex> l(_mutex);
+        if (!_queue.empty()) {
+            msg = _queue.front();
+            _queue.pop_front();
+            return true;
+        }
+        else
+        {
+            return false;
+        }
```

Comment:
`std::optional<T>` as return value?

```c++
msg = q.try_pop();
if (msg) { // or msg.has_value()
    // msg contains a (valid) value
} else {
    // msg does not contain a (valid) value
}

Reply:
I like it!  Never used that one before but it seems suitable.